interoperability_report.py

```diff
@@ -655,39 +682,53 @@ def main():
     now = datetime.now()
 
     for element in options['test_suite']:
-        t_suite_module = importlib.import_module(element,'./')
+        t_suite_module = importlib.import_module(element)
         for name, t_suite in inspect.getmembers(t_suite_module):
-            # we take all the dictionaries defined in the test_suite file except __builtins__.
+            # getmembers returns all the members in the t_suite_module.
+            # Then, 'type(t_suite) is a dict' takes all the members that
+            # are a dictionary. The only one that is not needed (it is not
+            # a test_suite) is __builtins__, and it is skipped.
             if type(t_suite) is dict and name != '__builtins__':
-                # check that the test_cases selected are in the test_suite
-                check_test_case_in_test_suite(t_suite, name, options['test_cases'])
-                check_test_case_in_test_suite(t_suite, name, options['test_cases_disabled'])
-
+                # check that the test_cases selected are in the test_suite and
+                # exit the application if they are not.
+                if (not are_tests_in_test_suite(t_suite, name, options['test_cases'])):
+                    return
+                if (not are_tests_in_test_suite(t_suite, name, options['test_cases_disabled'])):
+                    return
+
+                # k is the dictionary key and v the dictionary value
                 for k, v in t_suite.items():
                     # TestCase is a class from junitparser whose attributes
                     # are: name and result (OK, Failure, Error or Skipped).
-                    parameters = v[0]
-                    expected_codes = v[1]
-                    if len(v) == 3:
-                        check_function = v[2]
-                    elif len(v) == 2:
-                        check_function = no_check
+                    if (options['test_cases_disabled'] is not None and k in options['test_cases_disabled']):
```

Comment:
as these if are kind of complex, please add a comment saying in which situation we are
In this case, this if means "if there are test cases disabled and the script is processing one of them, continue"